File: engine/src/main/java/edu/austral/dissis/chess/engine/Player.java
package edu.austral.dissis.chess.engine;

import java.util.List;

public class Player {

  private boolean turn;
  public Color color;
  public List<Piece> pieces;
  public List<Piece> deadPieces;

  public Player(boolean turn, Color color, List<Piece> pieces, List<Piece> deadPieces) {
    this.turn = turn;
    this.color = color;
    this.pieces = pieces;
    this.deadPieces = deadPieces;
  }

  public Color getColor() {
    return color;
  }

  //public
}

File: engine/src/main/kotlin/edu/austral/dissis/chess/engine/game/Player.kt
package edu.austral.dissis.chess.engine.game

import edu.austral.dissis.chess.engine.piece.Color

class Player (private val color: Color) {

//  fun movePiece(board: Board, from: Position, to: Position): MovementResult {
//    val piece = board.getPiece(from) ?: throw IllegalArgumentException("No piece at $from")
//    if (piece.pieceValidator)
//    return ruleManager.validateMovement(board, movement)
//  }
  fun getColor(): Color {
    return color
  }
}